Update Godot to 3.6 RC 1 (for Flathub Beta)

Zishan-Rahman opened the pull request:
Using TuxFamily publish dates for release dates of 3.6 Betas 1 and 2 in update-appdata.patch, as they were each released a day before their respective blog articles were published

Beta 3's blog article and TuxFamily downloads were both published on the same date

Zishan-Rahman committed:
Update Godot to 3.6 Beta 3

Using TuxFamily publish dates for release dates of 3.6 Betas 1 and 2, as they were each released a day before their respective blog articles were published

Beta 3's blog article and TuxFamily downloads were both published on the same date

Zishan-Rahman commented:
Well, that's strange, the x86_64 build succeeded, but the aarch64 build didn't, and I'm not yet sure why. I'll try building it again.

bot, build

Zishan-Rahman commented:
I'm just wondering what's going wrong with the aarch64 build and why the x86_64 build isn't affected.

Zishan-Rahman commented:
Eh, I'll try it one more time, I guess.

bot, build

Zishan-Rahman commented:
@Calinou Sorry for the ping, but do _you_ have any idea why the x86_64 build succeeds but the aarch64 build doesn't? I'm open to your suggestions.

Calinou commented:
It looks like Embree is broken on ARM64… I guess you should pass `module_embree_enabled=no` in the SCons options (only when on ARM64). This will disable support for the CPU lightmapper on ARM64, but I doubt many people use it there anyway given ARM64 machines often have a slow CPU.

Zishan-Rahman commented:
I'll try building this again.

bot, build

Zishan-Rahman commented:
bot, build

Zishan-Rahman commented:
@Calinou (sorry for the ping) [I _did_ add `module_embree_enabled=no` to the SCons options for `aarch64` as per your instructions](https://github.com/flathub/org.godotengine.Godot3/pull/5/commits/2943634048461116535759c05ea8b6e60c687330), but the build still fails for that platform. It looks like it's still trying to build the Embree module for that platform anyway, even though the manifest is meant to instruct otherwise. Is it the way I _wrote_ the instruction(s) in the manifest? Or am I missing something else here? I'm open to your advice and suggestions.

Calinou commented:
I think you need to add `module_lightmapper_cpu_enabled=no` as well.

Zishan-Rahman commented:
Huh. I don't get it. It's _still_ trying to build Embree.

Zishan-Rahman commented:
I found a `builtin_embree` flag in the source code archive's `SConstruct` file so I'm using _that_ instead. Hopefully this time it works.

Zishan-Rahman commented:
```bash
In file included from modules/raycast/lightmap_raycaster.cpp:31:
modules/raycast/lightmap_raycaster.h:35:10: fatal error: embree3/rtcore.h: No such file or directory
   35 | #include <embree3/rtcore.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
In file included from modules/raycast/register_types.cpp:33:
modules/raycast/lightmap_raycaster.h:35:10: fatal error: embree3/rtcore.h: No such file or directory
   35 | #include <embree3/rtcore.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
```

```
scons: *** [modules/raycast/register_types.x11.opt.tools.64.o] Error 1
scons: *** [modules/raycast/lightmap_raycaster.x11.opt.tools.64.o] Error 1
scons: building terminated because of errors.
```

So I just checked the 3.6 beta 3 source code, and the raycast module relies on Embree for some reason, but not the CPU lightmapper module? Huh.

Well, I went ahead and disabled the raycast module. I'm pushing the change to this PR just to check whether or not it builds without the raycast module.

Zishan-Rahman commented:
[So this is the error](https://buildbot.flathub.org/#/builders/31/builds/6345/steps/8/logs/stdio):
```bash
Linking Program        ==> bin/godot.x11.opt.tools.64
/usr/lib/gcc/aarch64-unknown-linux-gnu/13.2.0/../../../../aarch64-unknown-linux-gnu/bin/ld: cannot find -lembree3: No such file or directory
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.x11.opt.tools.64] Error 1
scons: building terminated because of errors.
```
It looks like it's still trying to look for an `embree3` library at the library linking stage, even though we explicitly told it not to build `embree` on aarch64. And it _didn't_, so why it's still looking for it after the compile stage I don't know.

Well @Calinou (sorry for the ping) I tried everything I could try at the moment, but right now, I'm out of ideas. If you have any further suggestions I'm very much open to them.

Zishan-Rahman commented:
I'll refactor this PR (change name and commit history) to update to 3.6 Beta 4. Hopefully this doesn't have the same problems as before.

Zishan-Rahman committed:
Update Godot to 3.6 Beta 4

Zishan-Rahman committed:
Fix desktop-app-launchable-missing error

Zishan-Rahman committed:
Update Godot to 3.6 Beta 5

Using TuxFamily publish date (7th May 2024) instead of blog post date (13th May 2024)

Zishan-Rahman committed:
Fix appstream-missing-developer-name issue

Zishan-Rahman committed:
Fix metainfo-launchable-tag-wrong-value issue

Zishan-Rahman commented:
bot, build

Zishan-Rahman committed:
Use GitHub download link for Godot source code tarball while TuxFamily links are currently unavailable due to server downtime

Zishan-Rahman committed:
Update Godot to 3.6 RC 1

Zishan-Rahman committed:
Add arch=arm64 argument for aarch64 build to (hopefully) fix it

Zishan-Rahman commented:
Finally, this PR can be merged now. Thanks @akien-mga for [helping me fix the build problems](https://github.com/godotengine/godot/issues/94182) I've been having (and sorry for the ping).
